### PR TITLE
PX: detect hero images

### PR DESCRIPTION
**Pull request by sebastianbenz**
..and suggest using data-hero attribute

### CODE REVIEW

**Comment by sebastianbenz** [comment]
Hm. Looks like tests are non-deterministic as they depend on browser output.

File: packages/page-experience/lib/checks/heroImages.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+const isHeroImageCandidate = require('../helpers/isHeroImageCandidate');
+
+/**
+ * Checks if hero images are present in the initial viewport and declare data-hero
+ *
+ * @param {Object} pageData
+ */
+const earlyIframes = (pageData, result) => {
```
Comment (lluerich) [inline packages/page-experience/lib/checks/heroImages.js:24]: I guess this should be called heroImages or something more descriptive.

File: packages/page-experience/lib/checks/heroImages.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    (img) => !img.dataHero && isHeroImageCandidate(img)
+  );
+  if (heroImageCandidates.length === 0) {
+    return;
+  }
+
+  result.warn({
+    title: 'Improve LCP by optimizing hero images',
+    message:
+      'Let AMP Caches and Optimizers optimize your hero images, but adding the `data-hero` attribute to images in your first viewport.',
```
Comment (lluerich) [inline packages/page-experience/lib/checks/heroImages.js:35]: *"by adding"

**Review by lluerich — APPROVED** [review]
Two little nits and as you already mentioned the the one test relies on browser output and thus fails.

**Comment by sebastianbenz** [comment]
Thanks for the review! Fixed the nits.

**Review by lluerich — APPROVED** [review]
LGTM

**Pull request merged**